internal/consul/connection.go

```diff
@@ -81,6 +83,46 @@ func (c *client) Wait(until time.Duration) error {
 }
 
 func (c *client) WatchServers(ctx context.Context) error {
+	if !c.config.UseDynamic {
+		cfg := c.config.ApiClientConfig
+		cfg.Address = fmt.Sprintf("%s:%d", c.config.Addresses, c.config.HTTPPort)
```

Comment:
Not entirely specific to this line, but do we have unit test coverage for checking that both `http://` prefixed and plain IPs are handled for agentful and agentless paths?